### PR TITLE
MP-6939: Changes to encryptedtext type

### PR DESCRIPTION
**Pull request by danield-ap**

### CODE REVIEW

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, just small comments

File: src/Nox.EntityFramework.SqlServer/SqlServerEncryptedTextDatabaseConfigurator.cs
Comment (rochar) [inline src/Nox.EntityFramework.SqlServer/SqlServerEncryptedTextDatabaseConfigurator.cs:1]: removing the max and min, we should not need the SQL Configuration and we can let EF to do it by default to binary (please confirm its binary)

File: src/Nox.EntityFramework.Postgres/PostgresEncryptedTextDatabaseConfiguration.cs
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/PostgresEncryptedTextDatabaseConfiguration.cs:1]: we can also remove this, since we are removing Max and Min

**Review by rochar — APPROVED** [review]

File: src/Nox.Types/Types/EncryptedText/EncryptedText.cs [hunk 1]
```diff
@@ -11,9 +11,12 @@ public sealed class EncryptedText : ValueObject<byte[], EncryptedText>
 {
     public EncryptedText()
     {
-        Value = new byte[] { };
+        Value = new byte[] { };;
     }
 
+    /// <inheritdoc cref="FromEncryptedString"/>
+    public new static EncryptedText From(byte[] value) => FromEncryptedString(value);
```
[the remark this answers is not in the document]
Comment (danield-ap) [reply src/Nox.Types/Types/EncryptedText/EncryptedText.cs:18]: I think it depends on the use case. The most common scenario might be storing and retrieving data from a Database, if we are only planning on doing that it won't be necessary.
Other use case for encrypted text is communication. If this type will be used for sending data to another server/service, the other side will need to create the type from the ecnrypted byte array.

I removed the method for now.

File: src/Nox.Types/Types/EncryptedText/EncryptedText.cs [hunk 2]
```diff
@@ -44,7 +47,7 @@ public static EncryptedText FromPlainText(string value, EncryptedTextTypeOptions
     /// <param name="value">Encrypted data as byte array.</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static EncryptedText FromEncryptedTextBytes(byte[] value)
+    public static EncryptedText FromEncryptedString(byte[] value)
```
[the remark this answers is not in the document]
Comment (danield-ap) [reply src/Nox.Types/Types/EncryptedText/EncryptedText.cs]: Method removed.

File: src/Nox.Types.EntityFramework/Types/EncryptedText/EncryptedTextConverter.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+
+namespace Nox.Types.EntityFramework.Types;
+
+public class EncryptedTextConverter : ValueConverter<EncryptedText, byte[]>
+{
+    public EncryptedTextConverter() : base(encryptedText => encryptedText.Value, encryptedBase64 => EncryptedText.FromEncryptedString(encryptedBase64)) { }
```
[the remark this answers is not in the document]
Comment (danield-ap) [reply src/Nox.Types.EntityFramework/Types/EncryptedText/EncryptedTextConverter.cs]: Updated to use FromDatabase.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**